numpyro/infer/hmc.py

```diff
@@ -374,12 +382,38 @@ class HMC(MCMCKernel):
     :param float step_size: Determines the size of a single step taken by the
         verlet integrator while computing the trajectory using Hamiltonian
         dynamics. If not specified, it will be set to 1.
+    :param inverse_mass_matrix: Initial value for inverse mass matrix.
+        This may be adapted during warmup if adapt_mass_matrix = True.
+        If no value is specified, then it is initialized to the identity matrix.
+        For a potential_fn with general JAX pytree parameters, the order of entries
+        of the mass matrix is the order of the flatten version of pytree parameters,
+        which is a bit ambiguous (see more at
+        https://jax.readthedocs.io/en/latest/pytrees.html). If `model` is not None,
+        here we can specify a structured block mass matrix, with keys are tuple of
+        site names and values are the corresponding block of the mass matrix.
+        For more information about structured mass matrix, see `dense_mass` argument.
```

Comment:
nit:

a structured block mass matrix as a dictionary, where keys are tuples of
site names and values are the corresponding block of the mass matrix.
For more information about structured mass matrices, ...